specreduce/tracing.py

```diff
@@ -110,3 +113,164 @@ def __post_init__(self):
+        The image over which to run the trace. Assumes cross-dispersion
+        (spatial) direction is axis 0 and dispersion (wavelength)
+        direction is axis 1.
+    bins : int, optional
+        The number of bins in the dispersion (wavelength) into which to
+        divide the image. Use fewer if KosmosTrace is having
+        difficulty, such as with faint targets. Minimum bin size is 4.
+        [default: 20]
+    guess : int, optional
+        A guess at the trace's location in the cross-dispersion
+        (spatial) direction. If set, overrides the normal max peak
+        finder. Good for tracing a fainter source if multiple traces
+        are present. [default: None]
+    window : int, optional
+        Fit the trace to a region with size `window * 2` aroudn the
+        guess position. Useful for tracing faint sources if multiple
+        traces are present, but potentially bad if the trace is
+        substantially bent or warped. [default: None]
+
+    Improvements Needed
+    -------------------
+    1) switch to astropy models for Gaussian (done)
+    2) return info about trace width (?)
+    3) add re-fit trace functionality (or break off into other method)
+    4) add other interpolation modes besides spline, maybe via
+        specutils.manipulation methods?
+    """
+    image: CCDData
+    bins: int = 20
+    guess: float = None
+    window: int = None
+    disp_axis = 1
+    
+    def __post_init__(self,
+                      # Saxis=0, Waxis=1, display=False
+                      ):
+        if self.bins < 4:
+            raise ValueError('bins must be >= 4')
+        if not isinstance(self.bins, int):
+            warnings.warn('TRACE: Converting bins to int')
+            self.bins = int(self.bins)
+
+        if (self.window is not None
+            and (self.window > self.image.shape[self.disp_axis]
+                 or self.window < 1)):
+            raise ValueError("window must be >= 2 and less than the length of "
+                             "the image's spatial direction")
+        elif self.window is not None and not isinstance(self.window, int):
+            warnings.warn('TRACE: Converting window to int')
+            self.window = int(self.window)
+
+        # set max peak location by user choice or wavelength with max avg flux
+        ztot = np.nansum(self.image, axis=1) / self.image.shape[1]
+        peak_y = self.guess if self.guess is not None else np.nanargmax(ztot)
+        # (peak finder can be bad if multiple objects are on slit...)
+
+        # guess the peak width as the FWHM, roughly converted to gaussian sigma
+        yy = np.arange(len(ztot))
+        yy_above_half_max = np.sum(ztot > (np.nanmax(ztot) / 2))
+        width_guess = yy_above_half_max / 2.355
+
+        # enforce some (maybe sensible?) rules about trace peak width
+        width_guess = (2 if width_guess < 2
+                       else 25 if width_guess > 25
+                       else width_guess)
```

Comment:
While I'm all for intelligent software, I share @kecnry's concern about these hardcoded values. I'm wondering if these values can be extracted from some attribute of the image that's coming in? But that's outside my expertise

Reply:
I agree on the if/else clause, but the 2.355 in the first line you've highlighted comes from [this equation](https://en.wikipedia.org/wiki/Full_width_at_half_maximum#Normal_distribution) relating full width at half-maximum to standard deviation.